.github/workflows/test.yml

```diff
@@ -21,10 +23,8 @@ jobs:
           echo "------------------------------------------------------------------------"
           echo "Step 1: Install system dependencies"
           echo "------------------------------------------------------------------------"
-          sudo apt update
-          sudo apt dist-upgrade -y
```

Comment:
Good call to remove this. Doing this within a container defeats many of the things that make containers great anyway. If you need newer package deps, then base container should be adjusted for it